Mathlib/Analysis/Distribution/SchwartzSpace.lean

```diff
@@ -1215,4 +1247,112 @@ def toZeroAtInftyCLM : 𝓢(E, F) →L[𝕜] C₀(E, F) :=
+
+variable (𝕜 F) in
+/-- The `L^p` norm of a Schwartz function is controlled by a finite family of Schwartz seminorms.
+
+The maximum index `k` and the constant `C` depend on `p` and `μ`.
+-/
+theorem eLpNorm_le_seminorm (p : ℝ≥0∞) (μ : Measure E := by volume_tac)
+    [hμ : μ.HasTemperateGrowth] :
+    ∃ (k : ℕ) (C : ℝ≥0), ∀ (f : 𝓢(E, F)), eLpNorm f p μ ≤
+      C * ENNReal.ofReal ((Finset.Iic (k, 0)).sup (schwartzSeminormFamily 𝕜 E F) f) := by
+  -- Apply Hölder's inequality `‖f‖_p ≤ ‖f₁‖_p * ‖f₂‖_∞` to obtain the `L^p` norm of `f = f₁ • f₂`
+  -- using `f₁ = (1 + ‖x‖) ^ (-k)` and `f₂ = (1 + ‖x‖) ^ k • f x`.
+  rcases hμ.exists_eLpNorm_lt_top p with ⟨k, hk⟩
+  refine ⟨k, (eLpNorm (fun x ↦ (1 + ‖x‖) ^ (-k : ℝ)) p μ).toNNReal * 2 ^ k, fun f ↦ ?_⟩
+  have h_one_add (x : E) : 0 < 1 + ‖x‖ := lt_add_of_pos_of_le zero_lt_one (norm_nonneg x)
+  calc eLpNorm (⇑f) p μ
+  _ = eLpNorm ((fun x : E ↦ (1 + ‖x‖) ^ (-k : ℝ)) • fun x ↦ (1 + ‖x‖) ^ k • f x) p μ := by
+    refine congrArg (eLpNorm · p μ) (funext fun x ↦ ?_)
+    simp [Real.rpow_neg (h_one_add x).le, (h_one_add x).ne']
+  _ ≤ eLpNorm (fun x ↦ (1 + ‖x‖) ^ (-k : ℝ)) p μ * eLpNorm (fun x ↦ (1 + ‖x‖) ^ k • f x) ⊤ μ := by
+    refine eLpNorm_smul_le_eLpNorm_mul_eLpNorm_top p _ ?_
+    refine Continuous.aestronglyMeasurable ?_
+    exact .rpow_const (.add continuous_const continuous_norm) fun x ↦ .inl (h_one_add x).ne'
+  _ ≤ eLpNorm (fun x ↦ (1 + ‖x‖) ^ (-k : ℝ)) p μ *
+      (2 ^ k * ENNReal.ofReal (((Finset.Iic (k, 0)).sup (schwartzSeminormFamily 𝕜 E F)) f)) := by
+    gcongr _ * ?_
+    rw [eLpNorm_exponent_top]
+    refine eLpNormEssSup_le_of_ae_nnnorm_bound (ae_of_all μ fun x ↦ ?_)
+    rw [← norm_toNNReal, Real.toNNReal_le_iff_le_coe]
+    simpa [norm_smul, abs_of_nonneg (h_one_add x).le] using
+      one_add_le_sup_seminorm_apply (m := (k, 0)) (le_refl k) (le_refl 0) f x
+  _ = _ := by
+    rw [ENNReal.coe_mul, ENNReal.coe_toNNReal hk.ne]
+    simp only [ENNReal.coe_pow, ENNReal.coe_ofNat]
+    ring
+
+/-- The `L^p` norm of a Schwartz function is finite. -/
+theorem eLpNorm_lt_top (f : 𝓢(E, F)) (p : ℝ≥0∞) (μ : Measure E := by volume_tac)
+    [hμ : μ.HasTemperateGrowth] : eLpNorm f p μ < ⊤ := by
+  rcases eLpNorm_le_seminorm ℝ F p μ with ⟨k, C, hC⟩
+  exact lt_of_le_of_lt (hC f) (ENNReal.mul_lt_top ENNReal.coe_lt_top ENNReal.ofReal_lt_top)
+
+variable [SecondCountableTopologyEither E F]
+
+/-- Schwartz functions are in `L^∞`; does not require `hμ.HasTemperateGrowth`. -/
+theorem memℒp_top (f : 𝓢(E, F)) (μ : Measure E := by volume_tac) : Memℒp f ⊤ μ := by
+  rcases f.decay 0 0 with ⟨C, _, hC⟩
+  refine memℒp_top_of_bound f.continuous.aestronglyMeasurable C (ae_of_all μ fun x ↦ ?_)
+  simpa using hC x
+
+/-- Schwartz functions are in `L^p` for any `p`. -/
+theorem memℒp (f : 𝓢(E, F)) (p : ℝ≥0∞) (μ : Measure E := by volume_tac)
+    [hμ : μ.HasTemperateGrowth] : Memℒp f p μ :=
+  ⟨f.continuous.aestronglyMeasurable, f.eLpNorm_lt_top p μ⟩
+
+/-- Schwartz functions are in `L^p` for any `p`. -/
```

Comment:
The docstring gives the impression this is a theorem and not a definition. Can you adapt it?